mypy/nodes.py

```diff
@@ -895,12 +1062,30 @@ def __init__(self, name: str) -> None:
         self.name = name
         self.literal_hash = ('Var', name,)
 
-    def type_node(self):
-        return cast(TypeInfo, self.node)
-
     def accept(self, visitor: NodeVisitor[T]) -> T:
         return visitor.visit_name_expr(self)
 
+    def serialize(self) -> JsonDict:
```

Comment:
When do we need to serialize `NameExpr` objects? I couldn't find any examples where this would be needed, so perhaps it can be removed.